### PR TITLE
2020-12-03T20:09:06-06:00_FIX_IOSXR_Genie_Parser_ShowPlatform_&_Tested

### PR DESCRIPTION
**Pull request by bleuzkernel**
## Description
Hello again Team!,

Thanks in advance for taking the time to review my proposed changes to this code. I have encountered a bug parsing the commad "show platform" on a IOS-XR platform (The bug relies on the "ShowIPlatform" class becuase the code skips some line of the command output, on the line [376](https://github.com/CiscoTestAutomation/genieparser/blob/master/src/genie/libs/parser/iosxr/show_platform.py#L376) and [384](https://github.com/CiscoTestAutomation/genieparser/blob/master/src/genie/libs/parser/iosxr/show_platform.py#L384) only validates **0/0** specified in the regex, thus omitting the other slots). That behavior only occurs on certain platforms like Cisco CRS, ASR9k or NCS6k and it may not to work as expected.

#### CLI
```shell
RP/0/RP0/CPU0:Router#show platform 
Thu Dec  3 14:23:54.005 UTC
Node          Type              PLIM               State           Config State
------------- ----------------- ------------------ --------------- ---------------
0/0/CPU0      FP-140G           14-10GbE           IOS XR RUN      PWR,NSHUT,MON
0/1/CPU0      FP-140G           Services Plim      IOS XR RUN      PWR,NSHUT,MON
0/1/0         FP-140G(SPA)      CGSE-PLUS          OK              PWR,NSHUT,MON
0/2/CPU0      FP-140G           Services Plim      IOS XR RUN      PWR,NSHUT,MON
0/2/0         FP-140G(SPA)      CGSE-PLUS          OK              PWR,NSHUT,MON
0/3/CPU0      FP-X              4-100GbE           IOS XR RUN      PWR,NSHUT,MON
0/4/CPU0      FP-140G           Services Plim      IOS XR RUN      PWR,NSHUT,MON
0/4/0         FP-140G(SPA)      CGSE-PLUS          OK              PWR,NSHUT,MON
0/5/CPU0      FP-140G           Services Plim      IOS XR RUN      PWR,NSHUT,MON
0/5/0         FP-140G(SPA)      CGSE-PLUS          OK              PWR,NSHUT,MON
0/6/CPU0      FP-X              4-100GbE           IOS XR RUN      PWR,NSHUT,MON
0/7/CPU0      FP-X              40-10GbE           IOS XR RUN      PWR,NSHUT,MON
0/RP0/CPU0    RP(Active)        N/A                IOS XR RUN      PWR,NSHUT,MON
0/RP1/CPU0    RP(Standby)       N/A                IOS XR RUN      PWR,NSHUT,MON
```
#### Parsed Output
```json
{
    "slot": {
        "rp": {
            "0/RP0": {
                "name": "RP",
                "full_slot": "0/RP0/CPU0",
                "state": "IOS XR RUN",
                "config_state": "PWR,NSHUT,MON",
                "redundancy_state": "Active",
                "plim": "N/A"
            },
            "0/RP1": {
                "name": "RP",
                "full_slot": "0/RP1/CPU0",
                "state": "IOS XR RUN",
                "config_state": "PWR,NSHUT,MON",
                "redundancy_state": "Standby",
                "plim": "N/A"
            }
        }
    }
}
```
## Motivation and Context
My main motivation is to continue collaborating with the Cisco Dev Team.

## Impact (If any)
Not Apply.

#### Unit Test

``` shell
(sdk) λ saturn tests → λ git xr_sh_platform → python3 -m unittest discover                 
..................................................................................................................................................................................................................................................../home/bleuzkernel/DevNet/genieparsers/sdk/genieparser/src/genie/libs/parser/iosxr/show_controllers.py:78: FutureWarning: Possible nested set at position 74
  p2 = re.compile(r'^mac\=(?P<mac>[A-Fa-f0-9:]+) +vlan=(?P<vlan>\d+)'
/home/bleuzkernel/DevNet/genieparsers/sdk/genieparser/src/genie/libs/parser/iosxr/show_controllers.py:78: FutureWarning: Possible nested set at position 176
  p2 = re.compile(r'^mac\=(?P<mac>[A-Fa-f0-9:]+) +vlan=(?P<vlan>\d+)'
................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................../home/bleuzkernel/DevNet/genieparsers/sdk/genieparser/src/genie/libs/parser/utils/common.py:607: DeprecationWarning: This method will be removed in future versions.  Use 'list(elem)' or iteration over elem instead.
  cmd_node = root.getchildren()[0]
/home/bleuzkernel/DevNet/genieparsers/sdk/genieparser/src/genie/libs/parser/utils/common.py:620: DeprecationWarning: This method will be removed in future versions.  Use 'list(elem)' or iteration over elem instead.
  cmd_node = cmd_node.getchildren()
....................................................................../home/bleuzkernel/DevNet/genieparsers/sdk/genieparser/src/genie/libs/parser/nxos/show_bgp.py:8976: DeprecationWarning: This method will be removed in future versions.  Use 'list(elem)' or iteration over elem instead.
  for item in nei.getchildren():
/home/bleuzkernel/DevNet/genieparsers/sdk/genieparser/src/genie/libs/parser/nxos/show_bgp.py:8984: DeprecationWarning: This method will be removed in future versions.  Use 'list(elem)' or iteration over elem instead.
  root = item.getchildren()[0]
.....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
----------------------------------------------------------------------
Ran 1825 tests in 25.502s

OK
(sdk) λ saturn iosxr → λ git xr_sh_platform → pwd
/home/bleuzkernel/DevNet/genieparsers/sdk/genieparser/tests/iosxr
(sdk) λ saturn iosxr → λ git xr_sh_platform → python test_show_platform.py
.....................................................
----------------------------------------------------------------------
Ran 53 tests in 0.098s

OK
```

```python
import json
from genie.conf.base import Device

dev = Device(name='XR', os='iosxr')
dev.custom.abstraction = {'order': ['os']}
with open('show_platform_iosxr.txt', 'r') as f:
    output = f.read()
res = dev.parse('show platform', output=output)
print('expected_output =',json.dumps(res,indent=4))
```
> show_platform_iosxr.txt
```shell
RP/0/RP0/CPU0:Router#show platform 
Thu Dec  3 14:25:52.756 CST
Node          Type              PLIM               State           Config State
------------- ----------------- ------------------ --------------- ---------------
0/0/CPU0      FP-X              4-100GbE           IOS XR RUN      PWR,NSHUT,MON
0/1/CPU0      FP-X              4-100GbE           IOS XR RUN      PWR,NSHUT,MON
0/2/CPU0      FP-140G           Services Plim      IOS XR RUN      PWR,NSHUT,MON
0/2/0         FP-140G(SPA)      CGSE-PLUS          OK              PWR,NSHUT,MON
0/3/CPU0      FP-140G           Services Plim      IOS XR RUN      PWR,NSHUT,MON
0/3/0         FP-140G(SPA)      CGSE-PLUS          OK              PWR,NSHUT,MON
0/4/CPU0      FP-140G           Services Plim      IOS XR RUN      PWR,NSHUT,MON
0/4/0         FP-140G(SPA)      CGSE-PLUS          OK              PWR,NSHUT,MON
0/5/CPU0      FP-140G           Services Plim      IOS XR RUN      PWR,NSHUT,MON
0/5/0         FP-140G(SPA)      CGSE-PLUS          OK              PWR,NSHUT,MON
0/6/CPU0      FP-140G           Services Plim      IOS XR RUN      PWR,NSHUT,MON
0/6/0         FP-140G(SPA)      CGSE-PLUS          OK              PWR,NSHUT,MON
0/7/CPU0      FP-140G           Services Plim      IOS XR RUN      PWR,NSHUT,MON
0/7/0         FP-140G(SPA)      CGSE-PLUS          OK              PWR,NSHUT,MON
0/RP0/CPU0    RP(Active)        N/A                IOS XR RUN      PWR,NSHUT,MON
0/RP1/CPU0    RP(Standby)       N/A                IOS XR RUN      PWR,NSHUT,MON
```
>Output
```json
expected_output = {
    "slot": {
        "lc": {
            "0/0": {
                "name": "FP-X",
                "full_slot": "0/0/CPU0",
                "state": "IOS XR RUN",
                "config_state": "PWR,NSHUT,MON",
                "plim": "4-100GbE"
            },
            "0/1": {
                "name": "FP-X",
                "full_slot": "0/1/CPU0",
                "state": "IOS XR RUN",
                "config_state": "PWR,NSHUT,MON",
                "plim": "4-100GbE"
            },
            "0/2": {
                "name": "FP-140G",
                "full_slot": "0/2/CPU0",
                "state": "IOS XR RUN",
                "config_state": "PWR,NSHUT,MON",
                "plim": "Services Plim",
                "subslot": {
                    "0": {
                        "name": "FP-140G",
                        "state": "OK",
                        "config_state": "PWR,NSHUT,MON",
                        "redundancy_state": "SPA"
                    }
                }
            },
            "0/3": {
                "name": "FP-140G",
                "full_slot": "0/3/CPU0",
                "state": "IOS XR RUN",
                "config_state": "PWR,NSHUT,MON",
                "plim": "Services Plim",
                "subslot": {
                    "0": {
                        "name": "FP-140G",
                        "state": "OK",
                        "config_state": "PWR,NSHUT,MON",
                        "redundancy_state": "SPA"
                    }
                }
            },
            "0/4": {
                "name": "FP-140G",
                "full_slot": "0/4/CPU0",
                "state": "IOS XR RUN",
                "config_state": "PWR,NSHUT,MON",
                "plim": "Services Plim",
                "subslot": {
                    "0": {
                        "name": "FP-140G",
                        "state": "OK",
                        "config_state": "PWR,NSHUT,MON",
                        "redundancy_state": "SPA"
                    }
                }
            },
            "0/5": {
                "name": "FP-140G",
                "full_slot": "0/5/CPU0",
                "state": "IOS XR RUN",
                "config_state": "PWR,NSHUT,MON",
                "plim": "Services Plim",
                "subslot": {
                    "0": {
                        "name": "FP-140G",
                        "state": "OK",
                        "config_state": "PWR,NSHUT,MON",
                        "redundancy_state": "SPA"
                    }
                }
            },
            "0/6": {
                "name": "FP-140G",
                "full_slot": "0/6/CPU0",
                "state": "IOS XR RUN",
                "config_state": "PWR,NSHUT,MON",
                "plim": "Services Plim",
                "subslot": {
                    "0": {
                        "name": "FP-140G",
                        "state": "OK",
                        "config_state": "PWR,NSHUT,MON",
                        "redundancy_state": "SPA"
                    }
                }
            },
            "0/7": {
                "name": "FP-140G",
                "full_slot": "0/7/CPU0",
                "state": "IOS XR RUN",
                "config_state": "PWR,NSHUT,MON",
                "plim": "Services Plim",
                "subslot": {
                    "0": {
                        "name": "FP-140G",
                        "state": "OK",
                        "config_state": "PWR,NSHUT,MON",
                        "redundancy_state": "SPA"
                    }
                }
            }
        },
        "rp": {
            "0/RP0": {
                "name": "RP",
                "full_slot": "0/RP0/CPU0",
                "state": "IOS XR RUN",
                "config_state": "PWR,NSHUT,MON",
                "redundancy_state": "Active",
                "plim": "N/A"
            },
            "0/RP1": {
                "name": "RP",
                "full_slot": "0/RP1/CPU0",
                "state": "IOS XR RUN",
                "config_state": "PWR,NSHUT,MON",
                "redundancy_state": "Standby",
                "plim": "N/A"
            }
        }
    }
}
```
## Checklist:

- [x] I have updated the changelog.
- [x] I have updated the documentation (If applicable).
- [x] I have added tests to cover my changes (If applicable).
- [x] All new and existing tests passed.
- [ ] All new code passed compilation.

### CODE REVIEW

**Review by ahersi2-cisco — APPROVED** [review]

**Comment by ahersi2-cisco** [comment]
Looks good, just please address the comment above and you're good to merge after one more approval

**Comment by bleuzkernel** [comment]
Hi Ahersi,

I already performed the changes of the headers.

Thanks!
Ernesto Wilson

**Review by xiaoxinz-cisco — APPROVED** [review]

**Pull request merged**
